examples/react-signer/src/service/method/interactive/icrc34-get-delegation-method.service.ts

```diff
@@ -0,0 +1,114 @@
+import { RPCMessage, RPCSuccessResponse } from "../../../type"
+import { ComponentData, InteractiveMethodService } from "./interactive-method.service"
+import { Account, accountService } from "../../account.service"
+import { DelegationChain, Ed25519PublicKey } from "@dfinity/identity"
+import { Principal } from "@dfinity/principal"
+import { fromHex } from "@dfinity/agent"
+import { targetService } from "../../target.service"
+
+export interface AccountsComponentData extends ComponentData {
+  accounts: Account[]
+}
+
+export interface Icrc34Dto {
+  publicKey: string
+  targets: string[]
+  maxTimeToLive: string
+}
+
+class Icrc34GetDelegationMethodService extends InteractiveMethodService {
+  public getMethod(): string {
+    return "icrc34_get_delegation"
+  }
+
+  public async onApprove(message: MessageEvent<RPCMessage>, data?: unknown): Promise<void> {
+    const icrc34Dto = message.data.params as unknown as Icrc34Dto
+    const account = (data as Account[])[0]
+    const key = await accountService.getAccountKeyIdentityById(account.id)
+
+    if (!key) {
+      window.parent.postMessage(
+        {
+          origin: message.data.origin,
+          jsonrpc: message.data.jsonrpc,
+          id: message.data.id,
+          error: {
```

Comment:
Would be great to wrap errors into class.
As far as I remember - we have like 5 generic errors and that's it.

Reply:
Good catch! I wanted to make it within the story, but as far as Dan posted in X, we gonna show cool things on next session, I want firstly complete the logic and then refactor it.